### PR TITLE
feat: add buttons method

### PR DESCRIPTION
**Pull request by ahmadxgani**

### CODE REVIEW

**Review by hansputera — CHANGES_REQUESTED** [review]
We should check the qr file is exists or no.

**Comment by ahmadxgani** [comment]
i use else condition, so if !error, the file is exists, right? CMIIW

File: src/structures/context.ts [hunk 1]
```diff
@@ -108,6 +108,45 @@ export class Context {
     } else return undefined;
   }
 
+  /**
+   * Get detail from clicked basic button
+   *
+   * @return {proto.Message.IButtonsResponseMessage | undefined}
+   */
+  public get replyFromBasicButton():
```
Comment (hansputera) [inline src/structures/context.ts]: Can we change this function/method name to `clickedButtons` ?

Also, we could categorize or identify the clicked button to the `basic`, `template`, or `list` button.

File: src/utils/qrHandler.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ export const qrHandler = async (
     qrcode.toFile(qrOptions?.options.dest, qr, (err) => {
       if (err) console.error('Something was wrong:', err);
       else {
-        const pathFile = qrOptions.options!.dest as string;
+        const pathFile = qrOptions.options?.dest as string;
         if (fs.existsSync(path.resolve(pathFile))) {
```
Comment (hansputera) [inline src/utils/qrHandler.ts]: We should check the `pathFile` is exists here

File: src/structures/context.ts [hunk 3]
```diff
@@ -500,6 +577,37 @@ export class Context {
...
+  //         return this.sendRaw({ viewOnce: true, ...options }, { quoted: this.raw });
+  //     }
+  //   }
+
+  /**
+   * Reply a message using button template
+   *
+   * @param {AnyMessageContent} options Reply message options
+   */
+  public async replyWithButton(options: AnyMessageContent) {
```
Comment (hansputera) [inline src/structures/context.ts]: I think, for `replyWithButton` or `replyWithButtons` method, we should accept the `buttons`, and `options` params.

File: src/@typings/messages.ts [hunk 4]
```diff
@@ -13,6 +13,35 @@ export interface PollCreateEventData {
   pollId: string;
 }
 
+// interface BaseButtonOptions {
```
Comment (hansputera) [inline src/@typings/messages.ts]: Please remove unnecessary comments.

**Review by hansputera — CHANGES_REQUESTED** [review]

File: src/utils/qrHandler.ts (hunk 2 above)
Comment (ahmadxgani) [reply src/utils/qrHandler.ts]: wasn't it already checked using the existingSync method?

**Comment by hansputera** [comment]
I think, it would be better if you added a button builder to construct WhatsApp Button structures.

File: src/utils/qrHandler.ts (hunk 2 above)
Comment (hansputera) [reply src/utils/qrHandler.ts]: I mean, the value of  `pathFile` variable.

**Review by hansputera — CHANGES_REQUESTED** [review]
Please change the version of package to 1.6.9

**Review by hansputera — APPROVED** [review]
nice

**Pull request merged**